pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+
+    assert move_df.at[0, TRAJ_ID] == 1
+
+
+def test_values():
+    move_df = _default_move_df()
+
+    expected = [
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2]
+    ]
+    assert_array_equal(move_df.values, expected)
+
+
+def test_columns():
+    move_df = _default_move_df()
+
+    assert_array_equal(move_df.columns, [LATITUDE, LONGITUDE, DATETIME, TRAJ_ID])
+
+
+def test_index():
+    move_df = _default_move_df()
+
+    assert_array_equal(move_df.index, [0, 1, 2, 3])
+
+
+def test_dtypes():
+    move_df = _default_move_df()
+
+    expected = Series(
+        data=['float64', 'float64', '<M8[ns]', 'int64'],
+        index=['lat', 'lon', 'datetime', 'id'], dtype='object', name=None
+    )
+    assert_series_equal(move_df.dtypes, expected)
+
+
+def test_shape():
+    move_df = _default_move_df()
+
+    assert move_df.shape == (4, 4)
+
+
+def test_len():
+    move_df = _default_move_df()
+
+    assert move_df.len() == 4
+
+
+def test_unique():
+    move_df = _default_move_df()
+    assert_array_equal(move_df['id'].unique(), [1, 2])
+
+
+def test_head():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[0, 1]
+    )
+    assert_frame_equal(move_df.head(2), expected)
+
+
+def test_tail():
+    move_df = _default_move_df()
+
+    expected = DataFrame(data=[
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[2, 3]
+    )
+    assert_frame_equal(move_df.tail(2), expected)
+
+
+def test_number_users():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert move_df.get_users_number() == 1
+
+    move_df[UID] = [1, 1, 2, 3]
+    assert move_df.get_users_number() == 3
+
+
+def test_to_numpy():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_numpy(), ndarray)
+
+
+def test_to_dict():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_dict(), dict)
+
+
+def test_to_grid():
+    move_df = MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+    assert isinstance(move_df.to_grid(8), pymove.core.grid.Grid)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)